src/vaadin-combo-box-mixin.html

```diff
@@ -789,11 +789,17 @@
 
     _selectItemForValue(value) {
       const valueIndex = this._indexOfValue(value, this.filteredItems);
+      const previouslySelectedItem = this.selectedItem;
+
       this.selectedItem = valueIndex >= 0
         ? this.filteredItems[valueIndex]
         : (this.dataProvider && this.selectedItem === undefined)
           ? undefined
           : null;
+
+      if (this.selectedItem === null && previouslySelectedItem === this.selectedItem) {
```

Comment:
Can you cover these conditions with tests also? Or remove any conditions that isn't justified. The test in this pr currently passes fine without them.

Reply:
Done.

Reply:
The tests still pass without the first condition `this.selectedItem === null`!

Reply:
Fixed now @tomivirkki